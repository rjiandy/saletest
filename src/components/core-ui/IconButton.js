// @flow

import React, {Component} from 'react';
import {FontIcon} from 'material-ui';
import autobind from 'class-autobind';

type Props = {
    text?: string;
    icon: string;
    style?: {[key: string]: any};
    iconStyle?: {[key: string]: any};
    textStyle?: {[key: string]: any};
};

type State = {
    isHovered: boolean;
};

export default class IconButton extends Component<Props, State> {
    constructor() {
        super(...arguments);
        autobind(this);
        this.state = {
            isHovered: false,
        };
    }

    render() {
        let {text, icon, textStyle, style, iconStyle, ...otherProps} = this.props;
        let composedStyle = {...styles.iconButton, ...style};
        let composedIconStyle = {
            ...styles.icon, 
            color: this.state.isHovered ? 'rgb(200, 60, 60)' : 'rgba(100, 100, 100, 0.6)',
            ...iconStyle,
        };
        let composedTextStyle = {
            ...styles.text, 
            color: this.state.isHovered ? 'rgb(200, 60, 60)' : 'rgba(100, 100, 100, 0.6)',
            ...textStyle,
        };
        return (
            <div 
                style={composedStyle} 
                onMouseOver={() => this.setState({isHovered: true})} 
                onMouseOut={() => this.setState({isHovered: false})}
                {...otherProps}
            >
                <FontIcon className="material-icons" style={composedIconStyle}>{icon}</FontIcon>
                {
                    text ? (
                        <span style={composedTextStyle}>{text}</span>
                    ) : null
                }
            </div>
        )
    }
}

const styles = {
    iconButton: {
        display: 'flex',
        flex: 1,
        alignItems: 'center',
        paddingLeft: '20px',
        color: 'rgba(100, 100, 100, 0.6)',
        cursor: 'pointer',
    },
    icon: {
        fontSize: 16,
        color: 'rgba(100, 100, 100, 0.6)',
    },
    text: {
        marginLeft: '6px',
    },
};