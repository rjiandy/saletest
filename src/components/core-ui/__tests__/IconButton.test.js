import React from 'react';
import IconButton from '../IconButton';
import renderer from 'react-test-renderer';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

it('renders correctly with props', () => {
    let temp = (
        <MuiThemeProvider>
            <IconButton text="asd" icon="close" />
        </MuiThemeProvider>
    );
    const tree = renderer
      .create(temp)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });