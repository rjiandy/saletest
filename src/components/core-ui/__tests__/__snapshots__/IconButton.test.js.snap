// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props 1`] = `
<div
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  style={
    Object {
      "alignItems": "center",
      "color": "rgba(100, 100, 100, 0.6)",
      "cursor": "pointer",
      "display": "flex",
      "flex": 1,
      "paddingLeft": "20px",
    }
  }
>
  <span
    className="material-icons"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "color": "rgba(100, 100, 100, 0.6)",
        "display": "inline-block",
        "fontSize": 16,
        "muiPrepared": true,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "userSelect": "none",
      }
    }
  >
    close
  </span>
  <span
    style={
      Object {
        "color": "rgba(100, 100, 100, 0.6)",
        "marginLeft": "6px",
      }
    }
  >
    asd
  </span>
</div>
`;
